src/main/java/com/github/sardine/impl/SardineImpl.java

```diff
@@ -861,25 +868,25 @@ public void put(String url, InputStream dataStream, String contentType, boolean
 	@Override
 	public void put(String url, InputStream dataStream, String contentType, boolean expectContinue, long contentLength) throws IOException
 	{
-		InputStreamEntity entity = new InputStreamEntity(dataStream, contentLength);
+		InputStreamEntity entity = new InputStreamEntity(dataStream, contentLength, contentType != null ? ContentType.create(contentType) : null);
```

Comment:
Should we add a helper function since this is duplicated a bunch of times? Would be good to effectively document this.